### PR TITLE
Rationalize multiple origins

### LINKED ISSUE #441: Explain the functioning of origins in the challenge
There is an optional list of origins in the challenge, but the purpose of the field is not explained.

If I were to guess, the requirement to include the current server in the list means that clients are expected to retrieve tokens for that set of origins and -- if they receive extra -- use those tokens if challenged by any of the listed origins (provided that the other parameters match).

Section 2.1.2 almost says this, but there seems to be a missing link somewhere.

### PR DESCRIPTION
**Pull request by chris-wood**
Closes #441

### CODE REVIEW

File: draft-ietf-privacypass-auth-scheme.md [hunk 1]
```diff
@@ -229,7 +239,40 @@ The names MUST NOT include a "userinfo" portion of an authority. For example, a
 server name might be "issuer.example.com" or "issuer.example.com:8443",
 but not "issuer@example.com".
 
-### Sending Token Challenges
+#### Redemption Context Construction {#context-construction}
```
Comment (tfpauly) [inline draft-ietf-privacypass-auth-scheme.md:242]: Cool, thanks for moving up here. That makes sense.

File: draft-ietf-privacypass-auth-scheme.md [hunk 2]
```diff
@@ -220,7 +222,15 @@ multiple origin names, they are delimited with commas "," without any whitespace
 If this field is not empty, the Origin MUST include its own name as one of the
 names in the list.
 
-### Server Name Encoding {#server-name}
+If "origin_info" contains multiple origin names, this means the challenge is valid
+for any of the origins in the list, including the origin which issued the challenge
+(which must always be present in the list if it is non-empty; see {{process-challenge}}).
+This can be useful in settings where clients pre-fetch and cache tokens for a particular
+challenge -- including the "origin_info" field -- and then later redeem these tokens
+at one of the origins in the list. See {{caching}} for more discussion about
```
Comment (tfpauly) [inline draft-ietf-privacypass-auth-scheme.md:230]:
```suggestion
with one of the origins in the list. See {{caching}} for more discussion about
```

File: draft-ietf-privacypass-auth-scheme.md [hunk 3]
```diff
@@ -293,7 +336,7 @@ for one challenge over another (for example, if one uses a token type
 that is faster to verify), it can sort it to be first in the list
 of challenges as a hint to the client.
 
-### Process Token Challenges
+### Process Token Challenges {#process-challenge}
```
Comment (tfpauly) [inline draft-ietf-privacypass-auth-scheme.md:339]:
```suggestion
### Processing Token Challenges {#process-challenge}
```

Wondering if this sounds better

**Review by tfpauly — APPROVED** [review]

File: draft-ietf-privacypass-auth-scheme.md (hunk 3 above)
Comment (chris-wood) [reply draft-ietf-privacypass-auth-scheme.md:339]: Yep, it does!

**Pull request merged**